### PR TITLE
add exception lists

### PR DESCRIPTION
**Pull request by calkim-panther**
### Background

Add exception lists to rules to tune

### Testing

* Added tests using the Exception roles in test but not including with commit since the examples should be removed and would fail in customer instances.

### CODE REVIEW

File: rules/aws_cloudtrail_rules/aws_modify_cloud_compute_infrastructure.py [hunk 1]
```diff
@@ -59,15 +63,20 @@ def rule(event):
     # Disqualify any eventNames that do not Include instance
     # and events that have readOnly set to false
     if event.get("eventName", "") in EC2_CRUD_ACTIONS:
+        for allowed_arn in ALLOWED_ARNS:
+            useridentity_arn = deep_get(event, "userIdentity", "arn", default="<arn_not_found>")
```
Comment (edyesed) [inline rules/aws_cloudtrail_rules/aws_modify_cloud_compute_infrastructure.py:67]:
```suggestion
        useridentity_arn = deep_get(event, "userIdentity", "arn", default="<arn_not_found>")
        for allowed_arn in ALLOWED_ARNS:
```

File: rules/aws_cloudtrail_rules/aws_modify_cloud_compute_infrastructure.py [hunk 2]
```diff
@@ -59,15 +63,20 @@ def rule(event):
...
 
 
 def title(event):
-    items = deep_get(event, "requestParameters", "instancesSet", "items")
+    items = deep_get(event, "requestParameters", "instancesSet", "items", default=[{}])
     return (
         f"AWS Event [{event.get('eventName')}] Instance ID "
-        f"[{items[0].get('instanceId')}] AWS Account ID [{event.get('recipientAccountId')}]"
+        f"[{items[0].get('instanceId', '<instance_id_not_found>')}] "
+        f"mAWS Account ID [{event.get('recipientAccountId', '<account_id_not_found>')}]"
```
Comment (edyesed) [inline rules/aws_cloudtrail_rules/aws_modify_cloud_compute_infrastructure.py:79]: title should include the `actor`

**Review by edyesed — CHANGES_REQUESTED** [review]
I think that we should modify the approach of the ALLOWED_ARNS, and add some test cases showing that ALLOWED_ARNS do not raise alerts when ALLOWED_ARNS is configured

**Comment by calkim-panther** [comment]
> 

I did have a test case but something I remember occurring in the past is the detection gets shipped with a test case showing the sample allowed arn but when customers modify the allowlist the test case breaks. I was deciding between keeping the test case and leaving it up to the users to modify the test case after they modify the rule or just leaving it out altogether.

**Pull request closed without merging**